File: src/main/java/org/dukcode/ps/codetree/trail02/chapter03/lesson01/TestGetMedian2.java
package org.dukcode.ps.codetree.trail02.chapter03.lesson01;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/test-get-median-2/description">LINK</a>
 */
public class TestGetMedian2 {

  private static final int MX_LEN = 100;

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int currentLen = 0;
    int[] arr = new int[MX_LEN];

    int n = Integer.parseInt(br.readLine());

    st = new StringTokenizer(br.readLine());
    for (int i = 0; i < n; i++) {
      arr[i] = Integer.parseInt(st.nextToken());
      currentLen++;

      if (i % 2 == 0) {
        Arrays.sort(arr, 0, currentLen);
        bw.write(String.valueOf(arr[currentLen / 2]));
        bw.write(' ');
      }
    }

    br.close();
    bw.close();

  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter03/lesson01/TestGetMedian2.kt
package org.dukcode.ps.codetree.trail02.chapter03.lesson01.testGetMedian2

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/test-get-median-2/description">LINK</a>
 */
fun main() {
    val n = br.readLine().toInt()
    val arr = parseInts()

    val nums = mutableListOf<Int>()
    for (idx in 0 until n) {
        nums.add(arr[idx])

        if (idx % 2 == 0) {
            nums.sort()
            bw.write(nums[nums.size / 2].toString())
            bw.write(" ")
        }
    }

    bw.flush()
}

private fun parseInts() =
    StringTokenizer(br.readLine()).run {
        IntArray(countTokens()) {
            nextToken().toInt()
        }
    }
